api/handlers/refererhandler.py

```diff
@@ -136,6 +136,54 @@ def get(self, **kwargs):
         self.log_user_access(AccessType.view_container, cont_name=analysis['parent']['type'], cont_id=analysis['parent']['id'])
         return analysis
 
+    def get_all(self, cont_name, cid):
+
+        parent_to_child = {
+            'groups': 'projects',
+            'projects': 'sessions',
+            'sessions': 'acquisitions'
+        }
+
+        storages = {
+            'groups':       containerstorage.GroupStorage(),
+            'projects':     containerstorage.ProjectStorage(),
+            'sessions':     containerstorage.SessionStorage(),
+            'acquisitions': containerstorage.AcquisitionStorage()
+        }
+        parent_names = ['groups','projects','sessions', 'acquisitions']
+
+        # Only support groups/projects/sessions/acquisitions
+        if cont_name not in parent_names:
+            self.abort(400, "Analysis list not supported for {}".format(cont_name))
+
+        # Check that user has permission to container
+        container = storages[cont_name].get_container(cid)
+        permchecker = self.get_permchecker(container)
+        permchecker(noop)('GET')
+
+        parent_tree = {
+            cont_name: [cid]
+        }
+        parent_name = cont_name
+        while parent_to_child.get(parent_name):
+            # Parent storage
+            storage = storages[parent_name]
+            child_name = parent_to_child[parent_name]
+            parent_tree[child_name] = []
+
+            # For each parent id, find all of its children and add them to the list of child ids in the parent tree
+            for parent_id in parent_tree[parent_name]:
+                parent_tree[child_name] = parent_tree[child_name] + [cont["_id"] for cont in storage.get_children(parent_id, projection={'_id':1}, uid=self.uid)]
+
+            parent_name = child_name
+        # We only need a list of all the ids, no need for the tree anymore
+        parents = [pid for parent in parent_tree.keys() for pid in parent_tree[parent]]
+
+        # We set User to None because we check for permission when finding the parents
+        analyses = containerstorage.AnalysisStorage().get_all_el({'parent.id':{'$in':parents}},None,{'info': 0, 'files.info': 0})
```

Comment:
Because I think about this every time we rely on uniqueness of ObjectIds across collections (even though they are not guaranteed), I thought I would post it for reference: https://stackoverflow.com/questions/4677237/possibility-of-duplicate-mongo-objectids-being-generated-in-two-different-colle